ros/src/computing/perception/detection/packages/cv_tracker/nodes/yolo2/src/darknet/yolo2.cpp

```diff
@@ -0,0 +1,137 @@
+#include <string>
+#include <vector>
+
+#include <rect_class_score.h>
+
+extern "C"
+{
+	#undef __cplusplus
+		#include "detection_layer.h"
+		#include "parser.h"
+		#include "region_layer.h"
+		#include "utils.h"
+		#include "image.h"
+	#define __cplusplus
+}
+
+namespace darknet
+{
+	uint32_t Yolo2Detector::get_network_height()
+	{
+		return darknet_network_.h;
+	}
+	uint32_t Yolo2Detector::get_network_width()
+	{
+		return darknet_network_.w;
+	}
+	void Yolo2Detector::load(std::string& in_model_file, std::string& in_trained_file, double in_min_confidence, double in_nms_threshold)
+	{
+		min_confidence_ = in_min_confidence;
+		nms_threshold_ = in_nms_threshold;
+		darknet_network_ = parse_network_cfg(&in_model_file[0]);
+		load_weights(&darknet_network_, &in_trained_file[0]);
+		set_batch_network(&darknet_network_, 1);
+
+		layer output_layer = darknet_network_.layers[darknet_network_.n - 1];
+		darknet_boxes_.resize(output_layer.w * output_layer.h * output_layer.n);
+		darknet_box_scores_.resize(output_layer.w * output_layer.h * output_layer.n);
+		float *probs_mem = static_cast<float *>(calloc(darknet_box_scores_.size() * output_layer.classes, sizeof(float)));
+		for (auto& i : darknet_box_scores_)
+		{
+			i = probs_mem;
+			probs_mem += output_layer.classes;
+		}
+	}
+
+	Yolo2Detector::~Yolo2Detector()
+	{
+		free(darknet_box_scores_[0]);
+		free_network(darknet_network_);
+	}
+
+	std::vector< RectClassScore<float> > Yolo2Detector::detect(float *in_data)
+	{
+		return forward(in_data);
+	}
+
+	image Yolo2Detector::convert_image(const sensor_msgs::ImageConstPtr& msg)
+	{
+		if (msg->encoding != sensor_msgs::image_encodings::BGR8)
+		{
+			ROS_ERROR("Unsupported encoding");
+			exit(-1);
+		}
+
+		auto data = msg->data;
+		uint32_t height = msg->height, width = msg->width, offset = msg->step - 3 * width;
+		uint32_t i = 0, j = 0;
+		image im = make_image(width, height, 3);
+
+		for (uint32_t line = height; line; line--)
+		{
+			for (uint32_t column = width; column; column--)
+			{
+				for (uint32_t channel = 0; channel < 3; channel++)
+					im.data[i + width * height * channel] = data[j++] / 255.;
+				i++;
+			}
+			j += offset;
+		}
+
+		if (darknet_network_.w == (int) width && darknet_network_.h == (int) height)
+		{
+			return im;
+		}
+		image resized = resize_image(im, darknet_network_.w, darknet_network_.h);
+		free_image(im);
+		return resized;
+	}
+
+	std::vector< RectClassScore<float> > Yolo2Detector::forward(float *in_data)
+	{
+		float *prediction = network_predict(darknet_network_, in_data);
+		layer output_layer = darknet_network_.layers[darknet_network_.n - 1];
+
+		output_layer.output = prediction;
+		if (output_layer.type == DETECTION)
+			get_detection_boxes(output_layer, 1, 1, min_confidence_, darknet_box_scores_.data(), darknet_boxes_.data(), 0);
+		else if (output_layer.type == REGION)
+			get_region_boxes(output_layer, 1, 1, min_confidence_, darknet_box_scores_.data(), darknet_boxes_.data(), 0, 0, 0.5);
```

Comment:
It seems darknet's API of get_region_boxes was modificated by following commit:
https://github.com/pjreddie/darknet/commit/6731a3552d9c95280d7a27b18cea342751d79e3d

Could you follow up?
                                                                         ^